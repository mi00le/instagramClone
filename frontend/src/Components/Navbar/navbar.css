.disable-select {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}
.navbar-main{
  background-color: #ded8dd;
  background-image: linear-gradient(to right, #ded8dd, #ffffff);
}
.grid{
	color : #000;
}
.col-left{
	text-align: left;
}
.w3-animate-top{
animation-duration: 0.8s !important;
}

.popup{
	border : none;
	padding: 0px;
	border-radius: 20px;
}
.camera-icon{
	font-size: 40px;
	margin-top: 15px;
	color : #FFF;
	filter: drop-shadow( 1.8px 1.8px 1.8px #000 );

}
.camera-icon:active{
	filter: drop-shadow( 1px 1px 1px #000 );
}
.form-style{
	margin: 50px;
}
.input-style{

  	width: 100%;
  	text-align : center;
  	word-break: break-all;

}
.text-fix{
	overflow: auto; word-break: break-all;
}
.col-right{
	text-align: right;
}
a.insta-clone{
	text-decoration: none;
}
p.insta-clone{
	font-size: 40px;
	margin-top: 5px;
	margin-right: 31%;
	float: left;
	float: right;
	color: #FFF;
	font-family: 'Lobster', cursive;
	text-shadow: 1px 2px 3px black;
}
p.insta-clone:active{
	text-shadow: 1px 1px 2px black;
}
.form-container{

margin: 50px 50px 20px 20px;
text-align: right;

}

.flex-container{
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;


}
.flex-container > div:nth-child(odd){
  flex: 1 0 20%;
/* width: 20%; */
text-align: right;
padding-top: 20px;
margin: auto;

}
.flex-container > div:nth-child(even){
/* width: 80%; */
flex: 2 0 80%;
padding-top: 20px;

}




.popup-content{
border : none !important;
padding :  0px !important;
width : 60% !important;
min-width : 300px !important;
border-radius:10px !important;
background-color : #e7e7e7 !important;
max-width: 600px !important;
}
.btn-primary {
height : 50px !important;
float: right;
width: 80%;
}
#desc{
resize : none;
}
p.logout-nav{
	font-size: 40px;
	margin-top: 5px;
	float: right;
	color: #FFF;
	font-family: 'Lobster', cursive;
	text-shadow: 1px 2px 3px black;
}
p.logout-nav:active{

	text-shadow: 1px 1px 2px black;
}
p{
  word-wrap: break-word;
}
h3{
  word-wrap: break-word;
}

@media only screen and (max-width: 990px) {
 p.insta-clone {
    margin-right: 25%;
  }
}
@media only screen and (max-width: 767px) {
 p.insta-clone {
    visibility: hidden;
  }
}

input[type=text], textarea {
  width: 100%;
  padding: 12px;
  border: 1px solid #ccc;
  border-radius: 4px;
  resize: vertical;
}

label {
  padding: 12px 12px 12px 0;
  display: inline-block;
  padding-top: 0px;
}

.btn-primary {
  color: white;
  padding: 12px 20px;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  float: right;
  margin-top: 20px;
  width: 35%;
}

.btn-primary:hover {
  background-color: #2e6da4;
}

.form-container {
  border-radius: 5px;
  background-color: #f2f2f2;
  padding: 20px;
  margin: 20px;
}

.col-25 {
  float: left;
  width: 25%;
  margin-top: 6px;
  text-align: left;
}

.col-75 {
  float: left;
  width: 75%;
  margin-top: 6px;
}

/* Clear floats after the columns */
.row:after {
  content: "";
  display: table;
  clear: both;
}

/* Responsive layout - when the screen is less than 600px wide, make the two columns stack on top of each other instead of next to each other */
@media screen and (max-width: 600px) {
  .col-25, .col-75, .btn-primary {
    width: 100%;
    margin-top: 0;

  }
  label{
    padding-top: 5px;
  }
}
