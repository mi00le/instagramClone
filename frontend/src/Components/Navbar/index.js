import React from "react";
import "./navbar.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCamera } from "@fortawesome/free-solid-svg-icons";
// import Tags from "../../Components/Tags/index.js";

import Popup from "reactjs-popup";
import TagsInput from "react-tagsinput";
import "react-tagsinput/react-tagsinput.css";

import { Grid, Row, Col, Navbar as BSNavbar } from "react-bootstrap";

const tagLength = 10;

export default class Navbar extends React.Component {
  constructor() {
    super();
    this.state = {
      tags: [],
      tag: "",
      imageUrl: "",
      title: "",
      subject: ""
    };
  }

  handleChange = (tags) => {
    this.setState({ tags });
  }

  handleChangeInput = (tag) => {
    if(tag.length <= tagLength){
      this.setState({ tag });
    }
  }
  handleChangeImage = (event) => {
    this.setState({
      imageUrl: event.target.value
    })
  }
  handleChangeTitle = (event) => {
    this.setState({
      title: event.target.value
    })
  }
  handleChangeSubject = (event) => {
    this.setState({
      subject: event.target.value
    })
  }
  render() {
    return (
      <BSNavbar fixedTop collapseOnSelect className="navbar-main">
        <Grid className="grid">
          <Row className="show-grid">
            <Col xs={4} className="col-left">
              {this.props.isLoggedIn && (
                <Popup
                  className="w3-animate-top popup"
                  trigger={
                    <FontAwesomeIcon
                      href="#brand"
                      id="basic-nav-dropdown"
                      icon={faCamera}
                      className="camera-icon"
                    />
                  }
                  modal
                  closeOnDocumentClick
                >
                  <div className="form-container">
                    <div className="row">
                      <div className="col-25">
                        <label>Image</label>
                      </div>
                      <div className="col-75">
                        <input
                          type="text"
                          id="imgUrl"
                          className="form-control input-style"
                          placeholder="Image Url"
                          autoFocus
                          required
                        />
                      </div>
                    </div>
                    <div className="col-75">
                      <input
                        type="text"
                        onChange={this.handleChangeImage}
                        value={this.state.imageUrl}
                        className="form-control"
                        placeholder="Enter image url.."
                      />
                    </div>
                    <div className="row">
                      <div className="col-25">
                        <label>Tags</label>
                      </div>
                      <div className="col-75">
                        <TagsInput
                          maxTags="3"
                          value={this.state.tags}
                          onChange={this.handleChange.bind(this)}
                          inputValue={this.state.tag}
                          onChangeInput={this.handleChangeInput.bind(this)}
                        />
                      </div>
                    </div>
                    <div className="col-75">
                      <input
                        type="text"
                        onChange={this.handleChangeTitle}
                        value={this.state.title}
                        className="form-control"
                        placeholder="Enter a title.."
                      />
                    </div>
                    <div className="row">
                      <button
                        id="btn"
                        className="form-control btn-primary"
                        onClick={this.props.handler}
                      >
                        Upload
                      </button>
                    </div>
                  </div>
                  <div className="row">
                    <div className="col-25">
                      <label>Subject</label>
                    </div>
                    <div className="col-75">
                      <textarea
                        onChange={this.handleChangeSubject}
                        value={this.state.subject}
                        className="form-control"
                        name="subject"
                        placeholder="Write something.."
                        style={{ height: "200px" }}
                      />
                    </div>
                  </div>
                  <div className="row">
                    <button
                      id="btn"
                      className="form-control btn-primary"
                      onClick={() => {this.props.handler()}}
                    >
                      Upload
                    </button>
                  </div>
              </Popup>
              )}
            </Col>
            <Col xs={4} className="col-right">
              <a href="#top" className="insta-clone">
                <p
                  className={`disable-select ${
                    this.props.isLoggedIn ? "insta-clone" : "insta-cloneToggle"
                  }`}
                >
                  InstaClone
                </p>
              </a>
            </Col>
            <Col xs={4} className="col-right">
              {this.props.isLoggedIn && (
                <a
                  href="#logout"
                  className="logout-nav"
                  onClick={this.props.handleLogout}
                >
                  <p className="logout-nav disable-select ">Logout</p>
                </a>
              )}
            </Col>
          </Row>
        </Grid>
      </BSNavbar>
    );
  }
}
