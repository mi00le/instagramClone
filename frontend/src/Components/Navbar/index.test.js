
import React from 'react'
import Navbar from './'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

describe('Navbar', () => {
  it('should render', () => {
    const wrapper = shallow(<Navbar />)
    expect(wrapper).toBeTruthy()
  })

  it('should handle logout', () => {
    const handleLogout = jest.fn()
    const wrapper = shallow(<Navbar handleLogout={handleLogout} />)
    wrapper.find('a[href="#logout"]').simulate('click')
    expect(handleLogout).toHaveBeenCalled()
  })

  it('should match snapshot', () => {
    const tree = renderer
      .create(<Navbar />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
