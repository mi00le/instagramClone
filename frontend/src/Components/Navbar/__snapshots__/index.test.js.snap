// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should match snapshot 1`] = `
<nav
  className="navbar-main navbar navbar-default navbar-fixed-top"
>
  <div
    className="container"
  >
    <div
      className="grid container"
    >
      <div
        className="show-grid row"
      >
        <div
          className="col-left col-xs-4"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-camera fa-w-16 camera-icon"
            data-icon="camera"
            data-prefix="fas"
            href="#brand"
            id="basic-nav-dropdown"
            onClick={[Function]}
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 144v288c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V144c0-26.5 21.5-48 48-48h88l12.3-32.9c7-18.7 24.9-31.1 44.9-31.1h125.5c20 0 37.9 12.4 44.9 31.1L376 96h88c26.5 0 48 21.5 48 48zM376 288c0-66.2-53.8-120-120-120s-120 53.8-120 120 53.8 120 120 120 120-53.8 120-120zm-32 0c0 48.5-39.5 88-88 88s-88-39.5-88-88 39.5-88 88-88 88 39.5 88 88z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <div
          className="col-right col-xs-4"
        >
          <a
            className="insta-clone"
            href="#top"
          >
            <p
              className="insta-clone disable-select "
            >
              InstaClone
            </p>
          </a>
        </div>
        <div
          className="col-right col-xs-4"
        >
          <a
            className="logout-nav"
            href="#logout"
          >
            <p
              className="logout-nav disable-select "
            >
              Logout
            </p>
          </a>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
